### PR TITLE
Properly implement Brainstem.Collection#toServerJSON

### PR DESCRIPTION
**Pull request by jrolfs**
It now calls appropriate method on alls when serializing collection instead of deferring to Backbone method. This allows blacklisting to be applied to entire collections during serialization

### CODE REVIEW

**Comment by jrolfs** [comment]
@StephenGrider @juanca can you guys please take a look at this?

**Comment by jrolfs** [comment]
@cantino as well.

**Comment by StephenGrider** [comment]
why dont we just rename `toServerJSON` to `toJSON` and change it to 

`json = super
    blacklist = @defaultJSONBlacklist()

    switch method
      when "create"
        blacklist = blacklist.concat @createJSONBlacklist()
      when "update"
        blacklist = blacklist.concat @updateJSONBlacklist()

    for blacklistKey in blacklist
      delete json[blacklistKey]

    json`

then we can depend on backbone's default `toJSON`, which is 
`return this.map(function(model){ return model.toJSON(options); });`

**Comment by jrolfs** [comment]
`Backbone.Model#toJSON` is used for display internally in Marionette (and as a best practice in Backbone) so we can't apply the blacklist to that. It needs to be server specific.

**Comment by jrolfs** [comment]
A better implementation would probably have been to override `toJSON` and apply the blacklists based on an option but unfortunately that's a little bigger of a refactor than I want to bite off right now.

File: vendor/assets/javascripts/brainstem/brainstem-collection.coffee [hunk 1]
```diff
@@ -187,7 +189,7 @@ class window.Brainstem.Collection extends Backbone.Collection
     @_getCacheObject()?.valid = false
 
   toServerJSON: (method) ->
-    @toJSON()
+    @map (model) -> model.toServerJSON(method)
```
Comment (cantino) [inline vendor/assets/javascripts/brainstem/brainstem-collection.coffee]: `$.extend model.toServerJSON(method), id: model.id`

**Pull request merged**